origin-growth/src/apollo/resolvers.js

```diff
@@ -134,13 +134,18 @@ const resolvers = {
     },
     async enroll(_, args) {
       try {
-        return {
-          authToken: await authenticateEnrollment(
-            args.accountId,
-            args.agreementMessage,
-            args.signature
-          )
+        const authToken = await authenticateEnrollment(
+          args.accountId,
+          args.agreementMessage,
+          args.signature
+        )
+
+        // make referral connection after we are sure user provided the correct accountId
+        if (args.inviteCode !== undefined) {
+          await GrowthInvite.makeReferralConnection(args.inviteCode, args.accountId)
```

Comment:
I was thinking a bit more about this. What if the referee never enrolls to the growth program ? Are the onboarding flows highly encouraging any new user to enroll into growth program ? Otherwise the referrer will never get the reward. I think that's why we originally wanted to create this connection as part of the referee identity publish...

Reply:
hmmm, if the referee never enrols into growth I think there is no way for him to earn the tokens anyway. And there are some profile updates coming that should remind users about the possibility of earning the tokens:
https://app.zeplin.io/project/59fa2311bac7acbc8d953da9/screen/5c54aaee9c3e07307b3777df

I think the onboarding flow has similar small reminders about the possibility of earning tokens

Reply:
Right, if referee does not enroll there is no way he will get token rewards for attestation verification etc..

But my point was that it would also cause the referrer to not get the referral reward for that user, since there would be no referrer/referee connection recorded... I guess the question is "should referee enrolling in growth campaign be a condition for referrer to earn referral reward ?" With the implementation we have it is...

Reply:
Oh right I see your point. Yeah this is definitely not ideal, that referee also needs to enrol into growth. So our options: 
1. Store invite code in the identity IPFS. Pros: referee doesn't need to enrol into growth for referrer to get reward. Cons: Referee could update profile information before receiving an invite. In this case the code wouldn't make it to ipfs. 
2. Pass invite code to growth enrol. Pros: Profile/attestation events are attributed even if user doesn't update them after invite. Cons: Referee needs to be enrolled in order for referrer to claim reward.

Maybe we could make a third solution where:
- we store the invite code in localStorage when user reaches Welcome page
- we wait for the user to have Metamask installed and report the link between ethAddress and invite code to growth engine then? Once link has been reported we mark it in local storage, to not report the same invite code multiple times.

Reply:
I like your 3rd option... :)

Are you thinking there would be a new /referral_linking mutation that would get called by DApp2 when the accountID becomes available in DApp2 ? Is there a good place in DApp2 to make that call, like a method that gets called everytime it is detected the accountID changed ?

Reply:
I will just make a lightweight component that is present all the time, that will take care of the invite code reporting task